### ISSUE TITLE
Grouping API + payload_m creates unnecessary HNSW indexes

### ISSUE THREAD

**Issue by danielcrabtree** [qdrant/qdrant#2082]
Does the new Grouping API work well with Separate Partitions? 
I believe it creates unnecessary HNSW indexes that cannot currently be avoided through configuration.

Assume we have the following:
Collection `C` with payload fields `G` and `D`.
`C` is configured with payload_m of 16 and m of 0. 
A payload_index on `C` for field `G`.

Then we get separate partitions of the HNSW for each value of `G`.
This is as per the Separate partitions tutorial here - https://qdrant.tech/documentation/tutorials/multiple-partitions/

To use the Grouping API with field `D`, we add:
A payload_index on `C` for field `D`.

This causes the creation of separate HNSW indexes for each value of `D`.
But these indexes are not required in situations where `D` is only used for grouping.

@IvanPleshkov and @generall discussed the potential for this issue in #1257: "I propose to move payload_m from hnsw config to payload index config. Such API will be more complicated for users but it gets additional flexibility like different payload_m for graph tuning. For instance, user can skip hsnw index for some payload indexes." and "m is strictly the property of HNSW, it does not have meaning if we, for example, switch to another vector index."

This suggestion would appear to be a good way to resolve this issue. But if it is still desirable to keep the payload_m parameter as a property of HNSW rather than the payload index, then perhaps a new property on payload index to indicate if the vector index creation should be skipped for it.

**Comment by generall** [comment]
hi @danielcrabtree! You are correct, payload index on `D` will create links between "users".

Disabling additional HNSW index, however, might not be ideal solution for your case: if one "user" partition will be big enough, applying filter search to it may render HNSW graph disconnected, because additional links associated with `D` were not created.

What might be more accurate solution - is to allow creating joint payload indexes of multiple fields. In this case we could guarantee that 1) unnecessary links won't be created between groups, and 2) graph integrity within the group will be preserver. It also, theoretically, can be emulated with the current functionality by creating a concatenated field, since both `D` and `G` are keyword fields.

Do you have an estimation of how expensive it is for you to allow creation of the redundant links in HNSW?

**Comment by danielcrabtree** [comment]
I don't think that joint payload indexes or concatenating the fields helps. For this use case I never need to filter based on `D`, I'm only creating the index for the `group_by`.

Here is an example of the type of query I was intending to use:
```POST /collections/C/points/search/groups
{
  "vector": [0.1, 0.1, 0.9],
  "filter": {
    "must": [
      {
        "key": "G",
        "match": {
          "value": "user_1"
        }
      }
    ]
  },
  "group_by": "D",
  "limit": 4,
  "group_size": 2
}```

**Comment by danielcrabtree** [comment]
In that case, I expect the redundant links/graphs at least double the indexing and storage cost.

Before adding the index on `D`:
1) There is a connected component in the HNSW index for each "user" in `G`.

After adding the index on `D`:
2) Additional links/data are added to the connected components from 1 to support `group_by`.
3) Additional connected components are created in the HNSW index for each value in `D` to support filtering by `D`.
4) Additional links are created between the connected components in 1 and 3.

The cost of indexing and storage for 3 should be roughly equal to 1+2, but I assume only 1+2 are required to satisfy the earlier query.

4 also appears to be redundant for the earlier query.

Is this correct ?
